content/en/blog/_posts/2022-11-28-kubelet-credential-providers/index.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: blog
+title: 'Kubernetes v1.26: GA Support for Kubelet Credential Providers'
+date: 2022-11-28
+slug: kubelet-credential-providers
+---
+
+**Authors:** Andrew Sy Kim (Google), Dixita Narang (Google)
+
+Kubernetes v1.26 introduced generally available (GA) support for _kubelet credential
+provider plugins_, offering an extensible plugin framework to dynamically fetch credentials
+for any container image registry.
+
+## Background
+
+Kubernetes supports the ability to dynamically fetch credentials for a container registry service.
+Prior to Kubernetes v1.20, this capability was compiled into the kubelet and only available for
+Amazon Elastic Container Registry, Azure Container Registry, and Google Cloud Container Registry.
+
+{{< figure src="kubelet-credential-providers-in-tree.png" caption="Figure 1: Kubelet built-in credential provider support for Amazon Elastic Container Registry, Azure Container Registry, and Google Cloud Container Registry." >}}
+
+Kubernetes v1.20 introduced alpha support for kubelet credential providers plugins,
+which provide a mechanism for the kubelet to dynamically authenticate and pull images
+for arbitrary container registries - whether these are public registries, managed services,
+or even a self-hosted registry.
+In Kubernetes v1.26, this feature is now GA
+
+{{< figure src="kubelet-credential-providers-plugin.png" caption="Figure 2: Kubelet credential provider overview" >}}
+
+## Why is it important?
+
+Prior to Kubernetes v1.20,  if you wanted to dynamically fetch credentials for image registries
+other than ACR (Azure Container Registry), ECR (Elastic Container Registry), or GCR
+(Google Container Registry), you needed to modify the kubelet code.
+The new plugin mechanism can be used in any cluster, and lets you authenticate to new registries without
+any changes to Kubernetes itself. Any cloud provider or vendor can publish a plugin that lets you authenticate with their image registry.
+
+## How it Works
+
+The kubelet and the credential provider plugin binary communicate through stdio (stdin, stdout, and stderr) by sending and receiving
+json-serialized api-versioned types. If the exec plugin is enabled and the kubelet requires authentication information for an image
+that matches against a plugin, the kubelet will exec the plugin binary, passing the `CredentialProviderRequest` API via stdin. Then
+the exec plugin communicates with the container registry to dynamically fetch the credentials and returns the credentials in an
+encoded response of the `CredentialProviderResponse` API to the kubelet via stdout.
+
+{{< figure src="kubelet-credential-providers-how-it-works.png" caption="Figure 3: Kubelet credential provider plugin flow" >}}
+
+On receiving credentials from the Kubelet, the plugin can also indicate how long credentials can be cached for, to prevent unecessary
```

Comment:
Capitalization of kubelet not required.
```suggestion
On receiving credentials from the kubelet, the plugin can also indicate how long credentials can be cached for, to prevent unnecessary
```